### PR TITLE
avm2: Fix avmplus.describeTypeJSON bugs and add test

### PR DESCRIPTION
**Pull request by Aaron1011**

### CODE REVIEW

File: core/src/avm2/class.rs [hunk 1]
```diff
@@ -370,6 +372,7 @@ impl<'gc> Class<'gc> {
                     // A 'callable' class doesn't have a signature - let the
                     // method do any needed coercions
                     vec![],
+                    Multiname::any(activation.context.gc_context),
```
Comment (adrian17) [inline core/src/avm2/class.rs:375]: not PR related: yikes, we really should make `Any` not allocate at all, maybe replacing it by an Option-like enum. (IIRC avmplus uses a nullptr to represent an Any namespace and Any name)

File: core/src/avm2/globals/object.rs [hunk 2]
```diff
@@ -274,15 +274,43 @@ pub fn create_class<'gc>(activation: &mut Activation<'_, 'gc>) -> GcCell<'gc, Cl
...
-        ("propertyIsEnumerable", property_is_enumerable),
+    let as3_instance_methods: Vec<(&str, NativeMethodImpl, _, _)> = vec![
+        // These signatures are weird, but they match the describeTypeJSON output
+        (
+            "hasOwnProperty",
+            has_own_property,
+            vec![ParamConfig::optional(
+                "name",
+                Multiname::any(activation.context.gc_context),
+                true,
```
Comment (adrian17) [inline core/src/avm2/globals/object.rs:285]: wait, the default value of the param is `true`? not `undefined`?

**Review by adrian17 — APPROVED** [review]
Approved after clearing the question c:

File: core/src/avm2/globals/object.rs (hunk 2 above)
Comment (Aaron1011) [reply core/src/avm2/globals/object.rs]: oops - fixed

**Pull request merged**